### PR TITLE
test(robot): migrate test_replica_auto_balance_node_least_effort

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9776

#### What this PR does / why we need it:

migrate test_replica_auto_balance_node_least_effort

## Release Notes

- **New Features**
  - Introduced keywords for validating volume replica counts, including checks for specific nodes and ensuring no additional scheduling occurs.
  - Added methods to wait for a specific number of replicas for a volume across different classes, enhancing volume management capabilities.

- **Tests**
  - Implemented a new test case for validating replica auto-balancing behavior under specific conditions, ensuring proper distribution and health of volumes.

- **Chores**
  - Added a new resource for node-related keywords to support the new test case.

### CODE REVIEW

File: e2e/tests/regression/test_scheduling.robot [hunk 1]
```diff
@@ -51,3 +52,43 @@ Test Soft Anti Affinity Scheduling
     Then Wait until volume 0 replicas rebuilding completed
     And Wait for volume 0 healthy
     And Check volume 0 data is intact
+
+Test Replica Auto Balance Node Least Effort
```
Comment (khushboo-rancher) [inline e2e/tests/regression/test_scheduling.robot:56]: Can we add a brief description?

File: e2e/tests/regression/test_scheduling.robot [hunk 2]
```diff
@@ -51,3 +52,43 @@ Test Soft Anti Affinity Scheduling
...
+
+    When Enable node 1 scheduling
+    # wait for auto balance
+    Then Volume 0 should have replicas running on node 1
+    And Volume 0 should have 6 replicas running
+    # loop 3 times with 5-second wait and compare the replica count to:
+    # ensure no additional scheduling occurs
+    # the replica count remains unchanged
+    And Volume 0 should have replicas running on node 0 and no additional scheduling occurs
+    And Volume 0 should have replicas running on node 1 and no additional scheduling occurs
```
Comment (khushboo-rancher) [inline e2e/tests/regression/test_scheduling.robot:79]: Should we include the number of replicas as 3 running on node 0 & node 1 also in the verification?

File: e2e/tests/regression/test_scheduling.robot [hunk 3]
```diff
@@ -51,3 +52,43 @@ Test Soft Anti Affinity Scheduling
...
+    When Enable node 2 scheduling
+    # wait for auto balance
+    Then Volume 0 should have replicas running on node 2
+    And Volume 0 should have 6 replicas running
+    # loop 3 times with 5-second wait and compare the replica count to:
+    # ensure no additional scheduling occurs
+    # the replica count remains unchanged
+    And Volume 0 should have replicas running on node 0 and no additional scheduling occurs
+    And Volume 0 should have replicas running on node 1 and no additional scheduling occurs
+    And Volume 0 should have replicas running on node 2 and no additional scheduling occurs
```
Comment (khushboo-rancher) [inline e2e/tests/regression/test_scheduling.robot:91]: Same as above comment, including the replica count.

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]

File: e2e/tests/regression/test_scheduling.robot (hunk 2 above)
Comment (yangchiu) [reply e2e/tests/regression/test_scheduling.robot:79]: We've already check
```
And Volume 0 should have 6 replicas running
```
in the previous step.

File: e2e/tests/regression/test_scheduling.robot (hunk 3 above)
Comment (yangchiu) [reply e2e/tests/regression/test_scheduling.robot:91]: We've already check
```
And Volume 0 should have 6 replicas running
```
in the previous step.

File: e2e/tests/regression/test_scheduling.robot (hunk 1 above)
Comment (yangchiu) [reply e2e/tests/regression/test_scheduling.robot:56]: Added.

File: e2e/tests/regression/test_scheduling.robot (hunk 3 above)
Comment (yangchiu) [reply e2e/tests/regression/test_scheduling.robot]: Added.

File: e2e/tests/regression/test_scheduling.robot (hunk 2 above)
Comment (yangchiu) [reply e2e/tests/regression/test_scheduling.robot]: Added.

**Review by chriscchien — APPROVED** [review]

**Pull request merged**